src/librustc_trait_selection/traits/select.rs

```diff
@@ -2863,11 +2863,15 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             trait_ref,
         )?);
 
-        obligations.push(Obligation::new(
-            obligation.cause.clone(),
-            obligation.param_env,
-            ty::Predicate::ClosureKind(closure_def_id, substs, kind),
-        ));
+        // FIXME: Chalk
```

Comment:
we probably want to...track this a better way? I guess this code will all die by the time chalk merges

Reply:
Heh. This (and the changes in `canonicalizer`) are left around from the old implementation. I actually have no clue what these are for.

Reply:
not worth it